backend/cn/lib/check.ml

```diff
@@ -693,10 +693,15 @@ let rec check_pexpr (pe : BT.t mu_pexpr) (k : IT.t -> unit m) : unit m =
        (* in integers, perform this op and round. in bitvector types, just perform
           the op (for all the ops where wrapping is consistent) *)
        let@ () = WellTyped.WCT.is_ct act.loc act.ct in
-       assert (
+       let@ () =
          match act.ct with
-         | Integer ity when Sctypes.is_unsigned_integer_type ity -> true
-         | _ -> false);
+         | Integer ity when Sctypes.is_unsigned_integer_type ity -> return ()
+         | _ ->
+           fail (fun _ ->
+             { loc;
+               msg = Generic !^"M_PEbounded_binop: unsupported on non-integer types"
```

Comment:
This should probably mention that this is unsupported on anything other than *unsigned* integers.